feature/subscriber-attributes/src/main/java/com/revenuecat/purchases/subscriberattributes/SubscriberAttributesManager.kt

```diff
@@ -150,10 +150,8 @@ class SubscriberAttributesManager(
         appUserID: String,
         applicationContext: Application
     ) {
-        getDeviceIdentifiers(applicationContext) { deviceIdentifiers ->
-            val attributesToSet = mapOf(attributionKey.backendKey to value) + deviceIdentifiers
-            setAttributes(attributesToSet, appUserID)
-        }
+        setAttribute(attributionKey, value, appUserID)
+        collectDeviceIdentifiers(appUserID, applicationContext)
```

Comment:
This is an interesting solution, but I think attribution networks need the device identifiers in order to work properly. Like the backend needs `adjustID` but also the google identifiers. I am not sure about this, we should confirm this though

Reply:
for the vast majority of cases, I'd imagine that the time it takes to collect device ids is probably a lot slower than the time it takes to trigger an actual subs attributes sync, right?

So I imagine that we're looking to solve rare edge cases with this, where you call set ids and immediately start a purchase or immediately go to background. 

An ideal solution (albeit probably a big refactor) would be to make sure that backend calls happen _after_ any pending device identifiers requests. 

Thought: could we have those backend requests start on the same thread that collecting device ids happens on, so that they're  always guaranteed to run after existing ids collection?

Reply:
> Thought: could we have those backend requests start on the same thread that collecting device ids happens on, so that they're always guaranteed to run after existing ids collection?

Well, it's not the backend request that we need to use this thread on, but getting the unsynced subscriber attributes, which happens before. We can apply your idea to that though, which would fix this problem (without this issue I was mentioning). I'm slightly worried about overloading this thread with too much work... But then again, I don't think we are at a point where this is an issue just yet. Thanks for the idea!!  I like it more than this, so I will close this and do the new fix in a different PR.